app/models/manageiq/providers/vmware/infra_manager/refresh_parser_inventory_object.rb

```diff
@@ -82,7 +82,9 @@ def storage_inv_to_hashes(inv)
           )
         end
 
-        result << result.new_inventory_object(new_result)
+        new_result = result.new_inventory_object(new_result)
```

Comment:
That worked fine, what I wanted was to have the `InventoryObject` in the `result_uids` not the hash so that for example [this](https://github.com/ManageIQ/manageiq/pull/13185/commits/146505952b57cc51b29527532ce1ae72320329cc#diff-7dfcdfc48aae22ec920e79f6c125c6c2R952) would work right